etsd/keys/views.py

```diff
@@ -44,6 +47,34 @@ def get_context_data(self, *args, **kwargs):
 class PublicKeyDetailView(AdminOrAuthorityQsMixin, DetailView):
     model = models.PublicKey
 
+    def post(self, request, *args, **kwargs):
+        pubk= models.PublicKey.objects.get(pk=self.kwargs.get('pk'))
+
+        if "acceptKey" in request.POST:
+            activekeys = models.PublicKey.objects.filter(authority=pubk.authority, status="ACTIVE")
+            for key in activekeys:
+                key.status = "INACTIVE"
+                key.save()
+            pubk.approved_on = datetime.now()
+            pubk.status = "ACTIVE"
+        if "rejectKey" in request.POST:
+            pubk.status = "REJECTED"
+        
+        email_template = get_template("keys/emails/confirmation.txt")
+        email_ctx = dict(fingerprint=pubk.fingerprint, status = pubk.status,)
+        email_body = email_template.render(email_ctx)
+        usrs = users.models.User.objects.all()
+        recip_list = [usr.email for usr in usrs if usr.get_authority()==pubk.authority]
```

Comment:
Add `get_authority_users(authority)` and `get_authority_user_emails(authority)` functions to users.utils. Notice that the lines:

usrs = users.models.User.objects.all()
recip_list = [usr.email for usr in usrs if usr.get_authority()==pubk.authority]

are problematic: the usrs query will return *all* users and then you will filter them in the django side. This must be done in the db side. The proper way to do it is to use something like `authority.user_set.all()`.